openslides/users/static/js/users/site.js

```diff
@@ -850,76 +826,118 @@ angular.module('OpenSlidesApp.users.site', ['OpenSlidesApp.users'])
+                                 'motions': Motion.verboseName,
+                                 'users': User.verboseName };
+
+            var translation = translations[permissionApp];
+            if (translation) {
+                permissionApp = translation; // use translation if it is defined
+            } else {  // else: capitalize first char and use app name
+                permissionApp = permissionApp.charAt(0).toUpperCase() + permissionApp.slice(1);
+            }
+
+            if(permissionApp != lastApp) { // new cluster for the app of permission
+                var appCluster = { app_name: permissionApp,
+                                   app_visible: true,
+                                   permissions: [] };
+                lastApp = permissionApp;
+                $scope.apps.push(appCluster);
+            }
+
+            $scope.apps[$scope.apps.length - 1].permissions.push(perm);
+        });
+
+        // sort each app: first all permission with 'see', then 'manage', then the rest
+        // save the permissions in different lists an concat them in the right order together
+        angular.forEach($scope.apps, function (app) {
+            var see = [];
+            var manage = [];
+            var others = [];
+            angular.forEach(app.permissions, function (perm) {
+                if (perm.value.indexOf('see') > -1) {
+                    see.push(perm);
+                } else if (perm.value.indexOf('manage') > -1) {
+                    manage.push(perm);
+                } else {
+                    others.push(perm);
+                }
+            });
+            app.permissions = see.concat(manage.concat(others));
+        });
+
+        // check if the given group has the given permission
+        $scope.hasPerm = function (group, permission) {
+            return jQuery.inArray(permission.value, group.permissions) > -1;
+        };
+
+        // The current user is not allowed to lock himself out of the configuration:
+        // - if the permission is 'users.can_manage'
+        // - if the user is in 'group'
+        // - and if the group has the permission
+        $scope.danger = function (group, permission){
+            return (permission.value == "users.can_manage") &&
+                    operator.isInGroup(group) &&
+                    $scope.hasPerm(group, permission);
+        };
 
         // delete selected group
         $scope.delete = function (group) {
             Group.destroy(group.id);
         };
-    }
-])
 
-.controller('GroupCreateCtrl', [
-    '$scope',
-    '$state',
-    'Group',
-    'permissions',
-    function($scope, $state, Group, permissions) {
-        // get all permissions
-        $scope.permissions = permissions;
-        $scope.group = {};
-        $scope.save = function (group) {
-            if (!group.permissions) {
-                group.permissions = [];
+        // save changed permission
+        $scope.changePermission = function (group, perm) {
```

Comment:
I would prefer a code structure that does not contain inline `return` statements. The code might be much more readable.

Like this:
```javascript
if (!$scope.danger(group, perm)) {
    if (!$scope.hasPerm(group, perm)) {
        // activate perm
        group.permissions.push(perm.value);
    } else {
        // delete perm in group.permissions
        group.permissions = jQuery.grep(group.permissions, function(value) {
            return value != perm.value; // remove perm
        });
    }
   Group.save(group);     
}
```